### ISSUE TITLE
persistent-template's test suite fails to compile

### ISSUE THREAD

**Issue by jkachmar** [yesodweb/persistent#1020]
Issue duplicated from commercialhaskell/stackage#5111:

---
`persistent-template-2.8.2`'s test suite is failing to compile with multiple errors of the following form:
```
        • No instance for (Generic
                             (BackendKey Database.Persist.Sql.Types.Internal.SqlBackend))
            arising from the 'deriving' clause of a data type declaration
          Possible fix:
            use a standalone 'deriving instance' declaration,
              so you can specify the instance context yourself
        • When deriving the instance for (Generic (Key NoJson))
       |
    42 | share [mkPersist sqlSettings { mpsGeneric = False, mpsDeriveInstances = [''Generic] }, mkDeleteCascade 
sqlSettings { mpsGeneric = False }] [persistUpperCase|
       | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^...
```
---
I was looking around this repo and it looks like the failures I'm seeing on the Stackage builder are identical to [the ones cropping up in Travis for the last build](https://travis-ci.org/yesodweb/persistent/jobs/642978304?utm_medium=notification&utm_source=github_status).

**Comment by jkachmar** [comment]
I've skipped the `persistent-template` test suite on the Stackage Nightly builder for the time being, in order to keep everything running.

I'll try to keep an eye on this issue as well as the one on the Stackage issue tracker and re-add it later!

**Comment by parsonsmatt** [comment]
@jkachmar released as `persistent-template-2.8.2.1`, thanks!
